test/Parse/recovery.swift

```diff
@@ -44,15 +44,15 @@ func test(a: BadAttributes) -> () { // expected-note * {{did you mean 'test'?}}
 //===--- Recovery for braced blocks.
 
 func braceStmt1() {
-  { braceStmt1(); } // expected-error {{braced block of statements is an unused closure}} expected-error {{expression resolves to an unused function}}
+  { braceStmt1(); } // expected-error {{closure expression is unused}} expected-note {{did you mean to use a 'do' statement?}} {{3-3=do }}
 }
 
 func braceStmt2() {
-  { () in braceStmt2(); } // expected-error {{expression resolves to an unused function}}
+  { () in braceStmt2(); } // expected-error {{closure expression is unused}}
 }
 
 func braceStmt3() {
-  { // expected-error {{braced block of statements is an unused closure}}
+  {
```

Comment:
Why we want diagnostics here is: it's inconsistent with multi statements closure.
Consider:
```swift
func foo() {
  // ...
  {
    print("do it.")
    undefinedFunc()
  }
}
```
In this case, it's diagnosed as `closure expression is unused` `did you mean to use a 'do' statement?`.